cassandra/src/test/groovy/io/micronaut/configuration/cassandra/CassandraConfigurationSpec.groovy

```diff
@@ -35,86 +39,78 @@ class CassandraConfigurationSpec extends Specification {
 
         expect: "No beans are created"
         !applicationContext.containsBean(CassandraConfiguration)
-        !applicationContext.containsBean(Cluster)
+        !applicationContext.containsBean(CqlSession)
 
         cleanup:
         applicationContext.close()
     }
 
     void "test single cluster connection"() {
         given:
-        // tag::single[]
+        EmbeddedCassandraServerHelper.startEmbeddedCassandra();
         ApplicationContext applicationContext = new DefaultApplicationContext("test")
         applicationContext.environment.addPropertySource(MapPropertySource.of(
                 'test',
-                ['cassandra.default.clusterName': "ociCluster",
-                 'cassandra.default.contactPoint': "localhost",
-                 'cassandra.default.port': 9042,
-                 'cassandra.default.maxSchemaAgreementWaitSeconds': 20,
-                 'cassandra.default.ssl': true]
+                ['cassandra.datasource.default.basic.contact-points'                        : ["localhost:9142"],
+                 'cassandra.datasource.default.advanced.metadata.schema.enabled'            : false,
+                 'cassandra.datasource.default.basic.load-balancing-policy.local-datacenter': 'ociCluster']
         ))
         applicationContext.start()
-        // end::single[]
 
         expect:
-        !applicationContext.getBean(ClusterBuilderListener).invoked
+        !applicationContext.getBean(CqlSessionBuilderListener).invoked
         applicationContext.containsBean(CassandraConfiguration)
-        applicationContext.containsBean(Cluster)
+        applicationContext.containsBean(CqlSession)
 
         when:
-        Cluster cluster = applicationContext.getBean(Cluster)
-        applicationContext.getBean(ClusterBuilderListener).invoked
-        List<InetSocketAddress> inetSocketAddresses = cluster.manager.contactPoints
+        CqlSession session = applicationContext.getBean(CqlSession)
+        applicationContext.getBean(CqlSessionBuilderListener).invoked
+        Collection<LoadBalancingPolicy> policies = session.getContext().loadBalancingPolicies.values()
 
         then:
-        cluster.getClusterName() == "ociCluster"
-        inetSocketAddresses[0].getHostName() == "localhost"
-        inetSocketAddresses[0].getPort() == 9042
-        cluster.getConfiguration().getProtocolOptions().getMaxSchemaAgreementWaitSeconds() == 20
-        cluster.getConfiguration().getProtocolOptions().getSSLOptions() != null
+        ((DefaultLoadBalancingPolicy) policies[0]).getLocalDatacenter().get() == "ociCluster"
 
-
-        cleanup:
+        then:
+        EmbeddedCassandraServerHelper.cleanEmbeddedCassandra()
         applicationContext.close()
     }
 
     void "test multiple cluster connections"() {
         given:
-        // tag::multiple[]
```

Comment:
Same here as the `// tag::single[]`, this is linked in the documentation and we'll want to be aware of providing an alternative or keeping it in and modifying the surrounding text to support the new config.